server/server.go

```diff
@@ -1403,6 +1405,82 @@ func (s *Server) configureResolver() error {
 	return nil
 }
 
+// starts a file watcher to monitor for config file changes, will reload on changes
+func (s *Server) startWatchConf() error {
+	opts := s.getOpts()
+
+	// watch for file changes and issue a reload on changes
+	if opts.WatchConf {
+		if opts.ConfigFile == _EMPTY_ {
+			return fmt.Errorf("can't watch config file, no config file specified")
+		}
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return fmt.Errorf("can't create file watcher: %w", err)
+		}
+		s.mu.Lock()
+		s.configWatcher = watcher
+		s.mu.Unlock()
+
+		debouncer := debounce.NewDebouncer(200 * time.Millisecond)
+
+		go func() {
+			for {
+				select {
+				case <-s.quitCh:
+					s.configWatcher.Close()
+					return
+
+				case event, ok := <-s.configWatcher.Events:
+					if !ok {
+						return
+					}
+
+					// we watch the parent directory then filter
+					var found bool
+					if opts.ConfigFile == filepath.Base(event.Name) {
+						found = true
+					}
+					if !found {
+						continue
+					}
+
+					if event.Has(fsnotify.Write) || event.Has(fsnotify.Rename) {
```

Comment:
There are some extra things to do to handle the fsnotify events properly in some environments, this issue and PR to fix can offer some background:
- https://github.com/nats-io/nack/issues/130
- https://github.com/nats-io/nack/pull/131